### PR TITLE
Recording

### PR DESCRIPTION
**Pull request by frostbyte73**

### CODE REVIEW

File: livekit_models.proto [hunk 1]
```diff
@@ -67,3 +67,38 @@ message DataMessage {
...
+
+message RecordingTemplate {
+  string type = 1;
+  string ws_url = 2;
+  string token = 3;
+  string roomName = 4;
+}
+
+message RecordingOutput {
+  string file = 1;
```
Comment (davidzhao) [inline livekit_models.proto:88]: should this be a one_of?

File: livekit_models.proto [hunk 2]
```diff
@@ -67,3 +67,38 @@ message DataMessage {
     bytes binary = 2;
   }
 }
+
+message RecordingInput {
+  string url = 1;
```
Comment (davidzhao) [inline livekit_models.proto:72]: one_of?

File: livekit_room.proto [hunk 3]
```diff
@@ -37,6 +37,12 @@ service RoomService {
 
   // Subscribes or unsubscribe a participant from tracks. Requires `roomAdmin`
   rpc UpdateSubscriptions(UpdateSubscriptionsRequest) returns (UpdateSubscriptionsResponse);
+
+  // Begins recording a room
+  rpc RecordRoom(RecordRoomRequest) returns (RecordingResponse);
```
Comment (davidzhao) [inline livekit_room.proto:42]: are these required? or another way of recording?

**Review by davidzhao — APPROVED** [review]

File: livekit_room.proto (hunk 3 above)
Comment (frostbyte73) [reply livekit_room.proto:42]: another way, if you want to start it late or end it early

**Pull request merged**